### PR TITLE
go@1.21 1.21.7 (new formula)

### PR DESCRIPTION
**Pull request by stefanb**
Follow-up to
* https://github.com/Homebrew/homebrew-core/pull/157782 ( ⬅️  this has to be merged first, because it removes go@1.21 symlink alias ⚠️ )

Related:
* https://github.com/Homebrew/homebrew-core/pull/162668

---

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by chenrui333** [comment]
```
brew audit go@1.21 --new
brew test --verbose operator-sdk
brew test --verbose tfproviderlint
brew fetch --retry tkey-ssh-agent --build-bottle --force
```

there are some test failures with `operator-sdk` and `tfproviderlint`

**Comment by stefanb** [comment]
IMO we should handle the problematic 3 formulae separately from this PR. Should I move them to separate PRs?

**Comment by chenrui333** [comment]
> IMO we should handle the problematic 3 formulae separately from this PR. Should I move them to separate PRs?

yeah, you can. (I actually already did one)

File: Formula/c/clair.rb
Comment (chenrui333) [inline Formula/c/clair.rb:23]: we can also include the patch to ship with go1.22, no?

Comment (stefanb) [reply Formula/c/clair.rb:23]: Patching go.mod / go.sum is prone to conflicts, so we'd have to make a new patch specifically for the old release commit.

**Comment by stefanb** [comment]
Would it make sense to add tag `CI-no-fail-fast` ?

**Comment by chenrui333** [comment]
rebased and run after #157782 merge

**Comment by chenrui333** [comment]
We should remove `kapacitor` and `trezor-bridge` from the list (as they are not buildable from source at this point)

File: Formula/c/clair.rb
Comment (stefanb) [reply Formula/c/clair.rb:23]: There are > 100 commits since the last release, confirming my suspicion: https://github.com/quay/clair/compare/v4.7.2...main

**Review by p-linnane — APPROVED** [review]

**Comment by p-linnane** [comment]
13-x86_64 and 14-x86_64 are 404ing on the mirror, but it's definitely working:
```
╰─ curl -IL https://fossies.org/linux/misc/go1.21.7.src.tar.gz
HTTP/1.1 200 OK
```

**Comment by stefanb** [comment]
Any idea why would this be happening? This is not specifically related to Go, as this happens earlier, when trying to download the Go source tarball from the mirror.

File: Formula/g/go@1.21.rb
Comment (stefanb) [inline Formula/g/go@1.21.rb:5]: We could remove the `mirror` to get around this
```suggestion
```

Comment (p-linnane) [reply Formula/g/go@1.21.rb:5]: No idea why it's happening, may just be some network weirdness. I'm just going to merge this after CI finishes and not worry about it.

Comment (chenrui333) [reply Formula/g/go@1.21.rb:5]: shouldn't be a blocker. I can access the url fine.

**Pull request merged**
